Userland/Libraries/LibTLS/TLSv12.cpp

```diff
@@ -468,38 +470,37 @@ Vector<Certificate> TLSv12::parse_pem_certificate(ReadonlyBytes certificate_pem_
 Singleton<DefaultRootCACertificates> DefaultRootCACertificates::s_the;
 DefaultRootCACertificates::DefaultRootCACertificates()
 {
-    // FIXME: This might not be the best format, find a better way to represent CA certificates.
-    auto config_result = Core::ConfigFile::open_for_system("ca_certs");
-    if (config_result.is_error()) {
-        dbgln("Failed to load CA Certificates: {}", config_result.error());
+    auto cacert_result = Core::File::open("/etc/cacert.pem"sv, Core::File::OpenMode::Read);
+    if (cacert_result.is_error()) {
+        dbgln("Failed to load CA Certificates: {}", cacert_result.error());
         return;
     }
-    auto config = config_result.release_value();
-    reload_certificates(config);
+    auto cacert_file = cacert_result.release_value();
+    auto data_result = cacert_file->read_until_eof();
+    if (data_result.is_error()) {
+        dbgln("Failed to load CA Certificates: {}", data_result.error());
+        return;
+    }
+    auto data = data_result.release_value();
+    reload_certificates(data);
 }
 
-void DefaultRootCACertificates::reload_certificates(Core::ConfigFile& config)
+void DefaultRootCACertificates::reload_certificates(ByteBuffer& data)
 {
     m_ca_certificates.clear();
-    for (auto& entity : config.groups()) {
-        for (auto& subject : config.keys(entity)) {
-            auto certificate_base64 = config.read_entry(entity, subject);
-            auto certificate_data_result = decode_base64(certificate_base64);
-            if (certificate_data_result.is_error()) {
-                dbgln("Skipping CA Certificate {} {}: out of memory", entity, subject);
-                continue;
-            }
-            auto certificate_data = certificate_data_result.release_value();
-            auto certificate_result = Certificate::parse_asn1(certificate_data.bytes());
-            // If the certificate does not parse it is likely using elliptic curve keys/signatures, which are not
-            // supported right now. Currently, ca_certs.ini should only contain certificates with RSA keys/signatures.
-            if (!certificate_result.has_value()) {
-                dbgln("Skipping CA Certificate {} {}: unable to parse", entity, subject);
-                continue;
-            }
-            auto certificate = certificate_result.release_value();
-            m_ca_certificates.append(move(certificate));
+
+    auto certs = Crypto::decode_pems(data);
+
+    for (auto& cert_data : certs) {
+        auto certificate_result = Certificate::parse_asn1(cert_data.bytes());
+        // If the certificate does not parse it is likely using elliptic curve keys/signatures, which are not
+        // supported right now. It might make sense to cleanup cacert.pem before adding it to the system.
+        if (!certificate_result.has_value()) {
+            dbgln("Skipping CA Certificate: unable to parse");
```

Comment:
It would be nice if we could display details about the certificate that couldn't be parsed. Is this possible with the new architecture? Maybe you could have a 2-step parsing setup where the first step parses out the name in the certificate and the second step parses the signature. Completely optional of course.

Reply:
I though about that too, but I dont know how we would do that with what we currently have.